lib/go-tc/roles.go

```diff
@@ -59,8 +59,9 @@ type RoleV40 struct {
 
 // Validate will validate and make sure all that the fields in the supplied RoleV4 struct are semantically correct.
 func (role RoleV4) Validate() error {
+	noSpaces := validation.NewStringRule(tovalidate.NoSpaces, "cannot contain spaces")
 	errs := validation.Errors{
-		"name":        validation.Validate(role.Name, validation.Required),
+		"name":        validation.Validate(role.Name, validation.Required, noSpaces),
```

Comment:
Restricting names to not contain spaces is a breaking API change, and I don't think it's necessary to fix this bug.

Reply:
Should that validation also be removed from TP?